content/docs/getting-started/hello-features.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Tokio Features"
+weight : 1020
+menu:
+  docs:
+    parent: getting_started
+---
+
+Feel free to skip this section, and just use `features = ["full"]` while
+learning. This section provides some examples of how and when to specify
+tokio features. 
+
+## Often using all features is fine 
+
+When writing an app, we'll often use all of Tokio's features to accelerate
+our development time:
+
+```toml
+[dependencies]
+tokio = { version = "0.2", features = ["full"] }
+```
+
+For most apps, the additional compile time is hardly noticeable relative to the
+cognitive overhead of thinking about which features are needed.
+
+# Selecting features
+
+When we develop protocol implementations that are used by client apps and
+servers, we want to limit the dependencies to decrease compile time --
+not just for us, but also for the users of our crates.
+
+Let's take a look at our tiny client app and see what features are needed.
+
+## tokio main
+
+If we change our code, so that it only uses the `#[tokio::main]` macro:
+
+```
+#[tokio::main]
+async fn main() {
+    println!("doing nothing yet");
+}
+```
+
+then we can choose whether to use threads (commonly used for servers):
+
+```
+tokio = { version = "0.2", features = ["macros", "rt-threaded"] }
+```
+
+or lightweight cooperative multi-tasking
+(often required for low-profile clients): 
+
+```
+tokio = { version = "0.2", features = ["macros", "rt-core"] }
+```
+
+## TCPStream connect
+
+As we start to build the app, we'll need more features.  The line of code
+where we call `TcpStream::connect` requires two features: 
+
+
+```
+    let stream = TcpStream::connect("127.0.0.1:6142").await.unwrap();
+```
+
+
+1. `dns` for converting the string `127.0.0.1:6142` to an IP address -- Tokio 
+will need to do a DNS lookup just in case.
+2. `tcp` for handling the TCP connection
+
+
+## Writing to the socket
+
+Writing to the stream requires `io-util`:
+
+```
```

Comment:
```suggestion
```
	use tokio::io::AsyncWriteExt;
```

may want to add this and mention that `io-util` enables this import which then enables you to use `write`.

Reply:
I like `use tokio::prelude::*;` which I used in the simplified hello-world example (https://github.com/tokio-rs/website/pull/393/files) which then doesn't need `use tokio::io::AsyncWriteExt`

though probably would be good to mention what `io-util` enables -- I assume it is also AsyncReadExt?

Reply:
This may be more of a personal thing but I very very very much dislike preludes. I think it makes the benefit of code locality go away. Basically, if I am reading code and there are no `*` imports then I can resolve any local type to either being defined in the file or `use`ed at the top. To me when reading code this is super important. So for this reason I dislike using prelude. But up to you how you want to write this part :) That's just my personal 2cents.

Reply:
When first using tokio, I found it really difficult to construct code using `AsyncReadExt` and `AsyncWriteExt` (though maybe better docs would have helped!).  I do feel a little unfortable with the *magic* of prelude, but it does make the examples easier to write.

@LucioFranco can you say more about why you dislike preludes?  and why explicit `use` is a good thing and not just more typing?

Reply:
I added some text to provide a bit of a better explanation here.  As long as tokio has a prelude feature, I think hello-world example should use it, but agree that when we're talking about using *specific* features, people probably aren't looking to use prelude!